src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs

```diff
@@ -688,6 +722,96 @@ private static string GetSearchText(string parameterString)
             return parameterString.Substring(0, lastIndexOfSearchInSwitch);
         }
 
+        private string[] GetSelectedPackages()
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+
+            IntPtr hierarchyPtr = IntPtr.Zero;
+            IntPtr selectionContainerPtr = IntPtr.Zero;
+            try
+            {
+                if (!ErrorHandler.Succeeded(VsMonitorSelection.GetCurrentSelection(out hierarchyPtr, out uint itemId, out IVsMultiItemSelect multiItemSelect, out selectionContainerPtr))
+                    || (itemId == VSConstants.VSITEMID_NIL && hierarchyPtr != IntPtr.Zero))
+                {
+                    return null;
+                }
+
+                var packages = new List<string>();
+                if (itemId != VSConstants.VSITEMID_SELECTION)
+                {
+                    // This is a single item selection.
+
+                    IVsHierarchy hierarchy = Marshal.GetTypedObjectForIUnknown(hierarchyPtr, typeof(IVsHierarchy)) as IVsHierarchy;
+                    if (hierarchy != null)
+                    {
+                        if (ErrorHandler.Succeeded(hierarchy.GetProperty(itemId, (int)__VSHPROPID7.VSHPROPID_ProjectTreeCapabilities, out object treeCapabilities)))
+                        {
+                            string packageName = ParsePackageNameFromTreeCapabilities(treeCapabilities?.ToString());
+                            if (packageName != null)
+                            {
+                                packages.Add(packageName);
+                            }
+                        }
+                    }
+                }
+                else if (multiItemSelect != null)
+                {
+                    // This is a multiple item selection.
+
+                    uint numberOfSelectedItems;
+                    int isSingleHierarchyInt;
+                    if (ErrorHandler.Succeeded(multiItemSelect.GetSelectionInfo(out numberOfSelectedItems, out isSingleHierarchyInt)))
+                    {
+                        bool isSingleHierarchy = isSingleHierarchyInt != 0;
```

Comment:
If this is `false`, avoid the array allocation and call to `GetSelectedItems`. `hierarchy` will not be assigned anyway.